src/main/java/net/mcreator/ui/modgui/AttributeGUI.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2024, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.modgui;
+
+import net.mcreator.element.types.Attribute;
+import net.mcreator.ui.MCreator;
+import net.mcreator.ui.component.JMinMaxSpinner;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.help.HelpUtils;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.minecraft.EntityListField;
+import net.mcreator.ui.validation.AggregatedValidationResult;
+import net.mcreator.ui.validation.ValidationGroup;
+import net.mcreator.ui.validation.component.VTextField;
+import net.mcreator.ui.validation.validators.TextFieldValidator;
+import net.mcreator.util.StringUtils;
+import net.mcreator.workspace.elements.ModElement;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class AttributeGUI extends ModElementGUI<Attribute> {
+	private final VTextField name = new VTextField(20);
+	private final JSpinner defaultValue = new JSpinner(new SpinnerNumberModel(0.0, -Double.MIN_VALUE, Double.MAX_VALUE, 1.0));
+	private final JMinMaxSpinner minMaxValue = new JMinMaxSpinner(0, 1, -Double.MIN_VALUE, Double.MAX_VALUE, 1.0);
+	private final JCheckBox persists = L10N.checkbox("elementgui.common.enable");
+
+	private final EntityListField entities = new EntityListField(mcreator);
+	private final ValidationGroup page1group = new ValidationGroup();
+
+	public AttributeGUI(MCreator mcreator, ModElement element, boolean editingMode) {
+		super(mcreator, element, editingMode);
+		initGUI();
+		super.finalizeGUI();
+	}
+
+	protected void initGUI() {
+
+		JPanel pane1 = new JPanel(new BorderLayout());
+		JPanel pane2 = new JPanel(new GridLayout(5, 2, 5, 3));
+
+		pane1.setOpaque(false);
+		pane2.setOpaque(false);
+		persists.setOpaque(false);
+
+		minMaxValue.setPreferredSize(new Dimension(20, 20));
+		defaultValue.setPreferredSize(new Dimension(20, 20));
+
+		pane2.add(HelpUtils.wrapWithHelpButton(withEntry("common/gui_name"), L10N.label("elementgui.attribute.name")));

```

Comment:
This approach is not used in other codebase parts, so to keep code standardized, prefix withEntry with "this."

![image](https://github.com/user-attachments/assets/49b07a63-1081-4c4d-b3d3-fb78467e13f4)